segmentstore/server/src/main/java/io/pravega/segmentstore/server/security/TLSConfigChangeEventConsumer.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.security;
+
+import java.nio.file.WatchEvent;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * A {@link Consumer} that acts upon change in Segment Store SSL/TLS configuration change.
+ */
+@RequiredArgsConstructor
+@Slf4j
+public class TLSConfigChangeEventConsumer implements Consumer<WatchEvent<?>> {
+
+    /**
+     * A counter representing the number of times this object has been asked to
+     * consume an event.
+     */
+    private final AtomicInteger numOfConfigChangesSinceStart = new AtomicInteger(0);
+
+    private @NonNull final Channels channels;
+
+    @Override
+    public void accept(WatchEvent<?> watchEvent) {
+        log.debug("Invoked for [{}]", watchEvent.context());
+        handleTlsConfigChange();
+    }
+
+    private void handleTlsConfigChange() {
+        log.info("Current reload count = {}", numOfConfigChangesSinceStart.incrementAndGet());
+        channels.clear();
```

Comment:
I don't see the value in closing existing connections.  That said, an idle timeout on the connection would be handy to gracefully cleanup old connections.

Reply:
Closing existing channels will result in the client library code encountering connection drops. Upon connection drops, the library code automatically retries connecting to the server, and this time a new SslContext built using the latest certificate is used.

Channels can live for long. If they continue to use SslContext built from old certificates that have expired or are revoked, there shall likely be knotty issues to deal with w.r.t. error handling.

Reply:
@ravisharda your argument is not strong without an example of a 'knotty issue'.    I see more potential for knotty issues in closing the connection.

Once TLS negotiation is complete and the channel is active for data transfer, there is no practical reason to close the channel.

Reply:
Agree with Eron's comment. Consider it synonymous with filesystem calls, once you have file handle open for read/write it can be used even if the file has been renamed or deleted. File will actually be deleted only after the last file handle is closed.
Typically with expired certs problem is authentication, not the encryption they provide.
Since the session is already authenticated during the handshake before the cert was expired it should be good to use it as long as the connection is open.

Reply:
I've modified the implementation, as mentioned in a previous comment.